dp_consistency/README.md

```diff
@@ -1,54 +1,112 @@
+4. **Population Bound:** All adjusted estimates (Y_i) must be less than or equal to the total population (P).
+5. **Zero-Variance Equality:** Measurements with zero variance are considered exact and are enforced as equality
+   constraints (Y_i = M_j).
+
+## Interpretation
+
+* The variance adjustment in the objective function allows for larger changes to measurements with higher variance (
+  greater uncertainty) and encourages smaller changes to measurements with lower variance.
+* Under the assumption of normally distributed measurement noise, this approach is equivalent to finding the maximum
+  likelihood estimate (MLE) of the true values, meaning it selects the most likely consistent solution given the
+  observed data and their uncertainties.
+
+## Application to Origin's report
 
+### Measurements ###
 
-## How It Works
+Origin baseline measurements are either direct (single EDP measurements), or union measurements for more than one EDP.
+Each measurement is also associated to a time period, and a metric (e.g. mrc, ami, etc).
 
-1. **Data Representation:**
-   - Create a `SetMeasurementsSpec` object.
-   - Use the `add_subset_relation`, `add_cover`, and `add_measurement` methods to populate it with your data and the relationships between your sets.
-   - Use the (`Solver.solve`) method to remove inconsistencies in the measurements in (`SetMeasurementSpec`)
+To compute unique reach for each EDP (when the number of EDPs are larger than 2), we must also compute the union of all
+EDPs but one, for each EDP, for each measurement period and metric.
 
-2. **Constraint Generation:**
-   - The code automatically generates inequality and equality constraints based on the defined relationships.
-   - **Subset Constraints:** The sum of measurements in a subset must be less than or equal to the measurement of the parent set.
-   - **Cover Constraints:** The sum of measurements in the sets of a cover must be greater than or equal to the measurement of the parent set.
-   - **Zero-Variance Constraints:** Measurements with zero variance are treated as exact values, and equality constraints are added accordingly.
+### Cover Relationships ###
 
-3. **Quadratic Programming:**
-   - The objective function is the variance-adjusted L2 norm, which acts as a maximum likelihood estimate under the assumption of Gaussian noise.
-   - The `qpsolvers` library is used to solve the QP problem, finding the optimal solution that minimizes inconsistencies while respecting the constraints.
+For each metric and period, it identifies the set representing the union of all entities/dimensions (EDPs) and the
+individual
+_sets for each EDP and add a "cover" relationship to the spec, indicating that the union set's measurement should be_
+greater than or equal to the sum of the measurements of its constituent EDP sets.
 
-## Usage Example
+We do the same for the unions computed for unique reach purposes. Also including the relationship that the
```

Comment:
If we end up operating on the Report object vs. the sheet we may not have to deal with correcting unique reach because we can correct the n-way and (n-1)-way reaches.

Reply:
yeah this is what this sentence is trying to say, we operate corrections on the measured unions (not the derivatives)